File: FreeLine-iOS/Sources/Auth/AuthModels.swift
import Foundation

struct AuthTokens: Codable, Equatable {
    let accessToken: String
    let refreshToken: String
    let accessTokenExpiresAt: String
    let refreshTokenExpiresAt: String
}

struct AuthenticatedUser: Codable, Equatable {
    let id: String
    let email: String
    let displayName: String?
}

struct AuthSessionPayload: Codable, Equatable {
    let tokens: AuthTokens
    let user: AuthenticatedUser
}

struct PendingEmailVerification: Equatable {
    let email: String
    let previewLink: String
    let suggestedToken: String
}

enum DevAuthProvider: String {
    case apple
    case google

    var buttonTitle: String {
        switch self {
        case .apple:
            return "Continue with Apple (Dev)"
        case .google:
            return "Continue with Google (Dev)"
        }
    }

    var displayName: String {
        switch self {
        case .apple:
            return "Apple Dev"
        case .google:
            return "Google Dev"
        }
    }
}

enum AuthScreen {
    case welcome
    case email
}

File: FreeLine-Android/app/src/main/java/com/freeline/app/auth/AuthModels.kt
package com.freeline.app.auth

data class AuthTokens(
    val accessToken: String,
    val refreshToken: String,
    val accessTokenExpiresAt: String,
    val refreshTokenExpiresAt: String,
)

data class AuthenticatedUser(
    val id: String,
    val email: String,
    val displayName: String?,
)

data class AuthSessionPayload(
    val tokens: AuthTokens,
    val user: AuthenticatedUser,
)

data class PendingEmailVerification(
    val email: String,
    val previewLink: String,
    val suggestedToken: String,
)

enum class AuthScreen {
    Welcome,
    Email,
}

enum class DevAuthProvider(
    val routeName: String,
    val buttonTitle: String,
    val displayName: String,
) {
    Apple(
        routeName = "apple",
        buttonTitle = "Continue with Apple (Dev)",
        displayName = "Apple Dev",
    ),
    Google(
        routeName = "google",
        buttonTitle = "Continue with Google (Dev)",
        displayName = "Google Dev",
    ),
}
